controllers/dspipeline_controller.go

```diff
@@ -318,18 +319,18 @@ func (r *DSPAReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 				dspaStatus.SetScheduledWorkflowStatus, log)
 		}
 
-		err = r.ReconcileUI(dspa, params)
+		_, err = r.ReconcileUI(dspa, params)
 		if err != nil {
 			return ctrl.Result{}, err
 		}
 
-		err = r.ReconcileWorkflowController(dspa, params)
+		_, err = r.ReconcileWorkflowController(dspa, params)
 		if err != nil {
 			return ctrl.Result{}, err
 		}
 
 		// MLMD should be the last to reconcile because it can cause an early exit due to the lack of the TLS secret, which may not have been created yet.
-		err = r.ReconcileMLMD(ctx, dspa, params)
+		_, err = r.ReconcileMLMD(ctx, dspa, params)
```

Comment:
At least here we should use the returned status to set it in `r.setStatus`.

I wonder if we shouldn't use (set it in `r.setStatus`) the returned status in all cases when `status != nil`. The difference is that the condition will end up with a more friendly message instead of the original error. WDYT?

Reply:
As far as I can tell the setStatus functions all take actual metav1. conditions. How would we be using the returned status at all?